app/Http/Controllers/SessionController.php

```diff
@@ -33,18 +38,20 @@ public function __construct(ConfigFunctions $configFunctions, GitHubFunctions $g
 	/**
 	 * First function being called via AJAX.
 	 *
-	 * @return IlluminateResponse|array (array containing config information or killing the session)
+	 * TODO: Remove `status`; `status === 0 ` should result in a HTTP error response "no config", `status === {1|2}` should be a proper boolean `isAuthenticated`.
+	 *
+	 * @return array
+	 *
+	 * @throws ModelNotFoundException
+	 * @throws VersionControlException
 	 */
-	public function init()
+	public function init(): array
```

Comment:
This sound like something that @nagmat84 would refactor with a DTO. :)

Reply:
In theory, yes and I also gave it some thought but intentionally did not touch it given the prospect of its impact.

On the plus side, the returned array is luckily not used anywhere else (here we are at the outer boundary of the backend).

On the minus side, the returned values are wicked, undocumented and usually misnamed (especially all the values in the context of user authentication) and not type-safe. If I had started to write a DTO for that, this would have led to another large re-factoring not only of the backend but also of the frontend.

Before I touch this one, the user management needs to be repaired, the session management and then the updater/installer in that order. Then and only then it is worth considering cleaning up this messy array.